WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -148,13 +148,15 @@ final class AddEditCouponViewModel: ObservableObject {
          discountType: Coupon.DiscountType,
          stores: StoresManager = ServiceLocator.stores,
          storageManager: StorageManagerType = ServiceLocator.storageManager,
-         timezone: TimeZone = .siteTimezone) {
+         timezone: TimeZone = .siteTimezone,
+         onCompletion: ((Result<Coupon, Error>) -> Void)? = nil) {
```

Comment:
It makes sense  👍 I made it optional initially just for not declaring it in the unit tests, but it makes more sense to make it mandatory. Updated here 462bcc9e5ea9cf515d338e3b5199e3e26babf6a3

Reply:
Thanks! What do you think about my other suggestion though - should we return just the coupon instead of a `Result`? Since we expect `onCompletion` is triggered upon success 🤔